File: gems/decomplex/examples/php/locality-drag.php
<?php
function run($user, $cart, $logger) {
  $receipt_id = $user->id;

  $total = $cart->total;
  if ($total > 100) {
    if ($cart->discountable()) {
      $discount = 10;
    }
  }
  if ($cart->taxable()) {
    if ($cart->region) {
      $tax = $total * 0.2;
    }
  }
  if ($logger->enabled()) {
    if ($logger->debug()) {
      $logger->info($total);
    }
  }
  if ($cart->valid()) {
    if ($cart->ready()) {
      $status = READY;
    }
  }

  emit($receipt_id);
}

File: gems/decomplex/examples/zig/locality-drag.zig
pub fn run(user: User, cart: Cart, logger: Logger) void {
    const receipt_id = user.id;

    const total = cart.total;
    if (total > 100) {
        if (cart.discountable()) {
            const discount = 10;
            _ = discount;
        }
    }
    if (cart.taxable()) {
        if (cart.region) {
            const tax = total * 2;
            _ = tax;
        }
    }
    if (logger.enabled()) {
        if (logger.debug()) {
            logger.info(total);
        }
    }
    if (cart.valid()) {
        if (cart.ready()) {
            const status = 1;
            _ = status;
        }
    }

    emit(receipt_id);
}
